### PR TITLE
D8AGE-286: Scaffold the CDT submodule.

### PR DESCRIPTION
**Pull request by intelektron**
### Description

New submodule for CDT, with config requirements.

### Change log

- Added: The submodule, config requirements, test for requirements

### CODE REVIEW

File: modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+          'cdt.identifier_api_endpoint' => 'https://example.com/v2/requests/requestIdentifierByCorrelationId/:correlationId',
+          'cdt.status_api_endpoint' => 'https://example.com/v2/requests/:requestyear/:requestnumber',
+          'cdt.username' => 'test_username',
+          'cdt.client' => 'test_client',
+          'cdt.password' => 'test_password',
+        ],
+        [
+          'cdt.token_api_endpoint' => [
+            'value' => 'https://example.com/token',
+            'severity' => 0,
```
Comment (intelektron) [inline modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php]: Using constant here throws an type error, it seems like some PHPUnit issue.

File: modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  }
+
+  /**
+   * Data provider for testCdtRequirements.
+   *
+   * @return mixed[]
+   *   The test data.
+   */
+  public static function cdtRequirementsProvider(): array {
```
Comment (intelektron) [inline modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php]: I saw Francesco's comment about avoiding providers after creating this PR, I can change this test to eliminate the provider, if needed.

File: .drone.yml [hunk 3]
```diff
@@ -24,13 +24,25 @@ services:
       - EPOETRY_AUTH_CERT_PATH=""
       - EPOETRY_AUTH_CERT_PASSWORD=""
       - EPOETRY_AUTH_EU_LOGIN_BASEPATH=""
+      - CDT_TOKEN_API_ENDPOINT=https://example.com/token
```
Comment (drishu) [inline .drone.yml]: not sure when these example.com values will come into play, can we remove them and add anything only when needed

File: modules/oe_translation_cdt/oe_translation_cdt.install [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+use Drupal\Core\Site\Settings;
+
+/**
+ * Implements hook_requirements().
+ */
+function oe_translation_cdt_requirements($phase) {
+  $requirements = [];
+
+  if ($phase == 'runtime') {
+
```
Comment (drishu) [inline modules/oe_translation_cdt/oe_translation_cdt.install]: remove this newline, the if statement already demarcates the code block

File: modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    $requirements = oe_translation_cdt_requirements('runtime');
+    foreach ($settings as $name => $value) {
+      $this->assertArrayHasKey($name, $requirements);
+      if (in_array($name, ['cdt.password', 'cdt.api_key'])) {
+        $this->assertEquals('Value set', $requirements[$name]['value']);
+      }
+      else {
+        $this->assertEquals($value, $requirements[$name]['value']);
+      }
+      $this->assertEquals(0, $requirements[$name]['severity']);
```
Comment (drishu) [inline modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php]: can we use the REQUIREMENT_OK constant so its cleared

File: modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    // Test missing settings.
+    new Settings([]);
+    $requirements = oe_translation_cdt_requirements('runtime');
+    foreach ($settings as $name => $value) {
+      $this->assertArrayHasKey($name, $requirements);
+      $this->assertEquals('Value not set', $requirements[$name]['value']);
+      $this->assertEquals(1, $requirements[$name]['severity']);
```
Comment (drishu) [inline modules/oe_translation_cdt/tests/src/Kernel/RequirementsTest.php]: REQUIREMENT_WARNING

**Review by drishu — CHANGES_REQUESTED** [review]
Lets improve the PR title, something like Scaffold CDT submodule or Add ... or Create...

**Review by drishu — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**